### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by fan-ziqi**
# Description

- [add save_path to data_collect.py](https://github.com/leggedrobotics/viplanner/commit/9889c1f829849ecbf3d761a46b3aa219a734dafe)
- [fix readme script error](https://github.com/leggedrobotics/viplanner/commit/3fc7ed36d0096a720eb5465efcabc516f4be3a2f)
- [updated the import path for MatterportSemanticCostMapping from `nav_collectors.configs` to `omni.viplanner.config`](https://github.com/leggedrobotics/viplanner/commit/3e54ca2dc988644f05318d65349b778884d98b95) 
- [in headless mode, we cannot visualize the graph and omni.debug.draw is not available](https://github.com/leggedrobotics/viplanner/commit/bc5af14c093aaef6455910c4857956bd980e98d8) 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

Please attach before and after screenshots of the change if applicable.

## Checklist

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./formatter.sh`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by fan-ziqi** [comment]
Using headless mode can significantly improve the speed of collect data. Headless can be used normally in the matterport environment, but it cannot be directly used in the carla environment. The graphical interface must be run once in the Carla environment, and then stop the script and then can using headless mode. The problem is that `hits` is empty in following code:

https://github.com/leggedrobotics/viplanner/blob/7d481c1e0d1685a4e0258c8cdad5a5b7d41d1cb5/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py#L883-L910

error:
```bash
/home/ubuntu/anaconda3/envs/viplanner/lib/python3.10/site-packages/torch/functional.py:513: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3609.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
Traceback (most recent call last):
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/standalone/data_collect.py", line 136, in <module>
    main()
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/standalone/data_collect.py", line 118, in main
    samples = explorer.sample_viewpoints(args_cli.num_samples)
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/viewpoint_sampling.py", line 66, in sample_viewpoints
    self.terrain_analyser.analyse()
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py", line 74, in analyse
    self.construct_height_map()
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py", line 770, in construct_height_map
    hit_point = self._raycast_usd_stage(
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py", line 908, in _raycast_usd_stage
    hit_positions[hit_idx] = torch.tensor([single_hit["position"] for single_hit in hits if single_hit["hit"]]).to(
RuntimeError: shape mismatch: value tensor of shape [0] cannot be broadcast to indexing result of shape [0, 3]

```

File: omniverse/standalone/data_collect.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 """Launch Isaac Sim Simulator first."""
 
 import argparse
+import os
```
Comment (pascal-roth) [inline omniverse/standalone/data_collect.py:14]:
```suggestion
```

File: omniverse/standalone/data_collect.py [hunk 2]
```diff
@@ -54,6 +55,7 @@ def main():
     """Main function to start the data collection in different environments."""
     # setup sampling config
     cfg = ViewpointSamplingCfg()
+    cfg.save_path = os.path.join("${USER_SAVE_PATH}", args_cli.scene)
```
Comment (pascal-roth) [inline omniverse/standalone/data_collect.py:58]:
```suggestion
```

That should be defined in the config

**Comment by pascal-roth** [comment]
> Using headless mode can significantly improve the speed of collect data. Headless can be used normally in the matterport environment, but it cannot be directly used in carla and warehouse environment. The script must be run with non-headless mode in carla and warehouse environment once, and then stop the script and then can using headless mode. The problem is that `hits` is empty in following code:
> 
> https://github.com/leggedrobotics/viplanner/blob/7d481c1e0d1685a4e0258c8cdad5a5b7d41d1cb5/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py#L883-L910
> 
> error:
> 
> ```shell
> /home/ubuntu/anaconda3/envs/viplanner/lib/python3.10/site-packages/torch/functional.py:513: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3609.)
>   return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
> Traceback (most recent call last):
>   File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/standalone/data_collect.py", line 136, in <module>
>     main()
>   File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/standalone/data_collect.py", line 118, in main
>     samples = explorer.sample_viewpoints(args_cli.num_samples)
>   File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/viewpoint_sampling.py", line 66, in sample_viewpoints
>     self.terrain_analyser.analyse()
>   File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py", line 74, in analyse
>     self.construct_height_map()
>   File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py", line 770, in construct_height_map
>     hit_point = self._raycast_usd_stage(
>   File "/home/ubuntu/workspaces/viplanner_ws/viplanner/omniverse/extension/omni.viplanner/omni/viplanner/collectors/terrain_analysis.py", line 908, in _raycast_usd_stage
>     hit_positions[hit_idx] = torch.tensor([single_hit["position"] for single_hit in hits if single_hit["hit"]]).to(
> RuntimeError: shape mismatch: value tensor of shape [0] cannot be broadcast to indexing result of shape [0, 3]
> ```
> 
> I don't know how to fix this. After fixing this issue, the `data_collect.py` script should run in headless mode to accelerate collection.

Good catch, I am not entirely sure. Moving it now to an issue to keep track of it.

**Pull request merged**